src/pymor/reductors/stokes.py

```diff
@@ -0,0 +1,146 @@
+
+        if self.projection_method == 'supremizer_galerkin':
+            # Enrich velocity basis RB_u with one supremizer per pressure basis vector:
+            # len(RB_u) <- len(RB_u) + len(RB_p)
+            with self.logger.block(
+                f'Enriching RB_u with supremizers: len(RB_u) += len(RB_p) = {len(RB_p)}'
+            ):
+                supremizers = self.compute_supremizers()
+                self.extend_basis(supremizers, 'RB_u')
+
+            # build V_block again as it changed due to supremizer enrichment
+            V_block = fom.solution_space.make_block_diagonal_array((self.bases['RB_u'], RB_p))
+            projected_operators = StationaryRBReductor(fom, RB=V_block, check_orthonormality=False).project_operators()
+
+        elif self.projection_method == 'ls-normal' or self.projection_method == 'ls-ls':
+            if self.projection_method == 'ls-normal':
+                projected_operators = StationaryLSRBReductor(fom, RB=V_block, product=mixed_product,
+                                                             use_normal_equations=True,
+                                                             check_orthonormality=False).project_operators()
+            else:
+                projected_operators = StationaryLSRBReductor(fom, RB=V_block, product=mixed_product,
+                                                             check_orthonormality=False).project_operators()
+
+        else:
+            raise NotImplementedError
+
+        return projected_operators
+
+    def compute_supremizers(self):
+        fom = self.fom
+        RB_p = self.bases['RB_p']
+        u_product = fom.u_product
+
+        block_pu = fom.operator.blocks[1,0]
+        supremizers = fom.solution_space.subspaces[0].empty()
+
+        supremizer_rhs = block_pu.apply_adjoint(RB_p)
+        if u_product:
+            supremizer_vector = u_product.apply_inverse(supremizer_rhs)
+        else:
+            supremizer_vector = supremizer_rhs
+
+        norm = supremizer_vector.norm()
+        supremizer_vector.scal(1 / norm)
+        supremizers.append(supremizer_vector)
+
+        return supremizers
+
+    def project_operators_to_subbasis(self, dims):
+        rom = self._last_rom
+        dim_u = dims['RB_u']
+        dim_p = dims['RB_p']
+        dim_trial = dim_u + dim_p
+
+        if self.projection_method == 'supremizer_galerkin' or self.projection_method == 'ls-normal':
+            # Square system: both dimensions are the same
+            projected_operators = {
+                'operator':          project_to_subbasis(rom.operator, dim_trial, dim_trial),
+                'rhs':               project_to_subbasis(rom.rhs, dim_trial, None),
+                'products':          {k: project_to_subbasis(v, dim_trial, dim_trial)
+                                      for k, v in rom.products.items()},
+                'output_functional': project_to_subbasis(rom.output_functional, None, dim_trial)
+            }
+
+        else:
```

Comment:
I think you want to save the `StationarayLSRBReductor` as an attribute and call its `project_operator_to_subasis` here.

Reply:
I changed it, but it only works if `project_operators_to_subbasis` in the `StationaryLSRBReductor` gets an additional, optional argument `last_rom` (the `StationaryLSRBReductor` itself has no `_last_rom`, as `reduce` (where this is initialized) gets called on the `StationaryRBStokesReductor` by the user).